server/authn/config/config.go

```diff
@@ -0,0 +1,54 @@
+// Copyright AGNTCY Contributors (https://github.com/agntcy)
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"errors"
+	"fmt"
+)
+
+// AuthMode specifies the authentication mode (jwt or mtls).
+type AuthMode string
+
+const (
+	AuthModeJWT  AuthMode = "jwt"
+	AuthModeMTLS AuthMode = "mtls"
+)
+
+// Config contains configuration for authentication services.
+type Config struct {
+	// Indicates if authentication is enabled
+	Enabled bool `json:"enabled,omitempty" mapstructure:"enabled"`
+
+	// Authentication mode: "jwt" or "mtls"
+	Mode AuthMode `json:"mode,omitempty" mapstructure:"mode"`
+
+	// SPIFFE socket path for authentication
+	SocketPath string `json:"socket_path,omitempty" mapstructure:"socket_path"`
+
+	// Expected audiences for JWT validation (only used in JWT mode)
+	Audiences []string `json:"audiences,omitempty" mapstructure:"audiences"`
+}
+
+func (c *Config) Validate() error {
+	if !c.Enabled {
+		return nil
+	}
+
+	if c.SocketPath == "" {
+		return errors.New("socket path is required")
+	}
+
+	// Validate mode
+	if c.Mode != AuthModeJWT && c.Mode != AuthModeMTLS {
+		return fmt.Errorf("invalid auth mode: %s (must be 'jwt' or 'mtls')", c.Mode)
+	}
+
+	// JWT-specific validation
+	if c.Mode == AuthModeJWT && len(c.Audiences) == 0 {
+		return errors.New("at least one audience is required for JWT mode")
+	}
```

Comment:
nit: separate validation based on authn type

```suggestion
	if !c.Enabled {
		return nil
	}

	// Switch and validate based on type
	switch c.Mode {
		case AuthModeJWT:
			if len(c.Audiences) == 0 {
				return errors.New("at least one audience is required for JWT mode")
			}
			
			return nil

		case AuthModeMTLS:
			if c.SocketPath == "" {
				return errors.New("socket path is required")
			}
			
			return nil
			
		// unsupported mode
		default:
			return fmt.Errorf("unsupported authn mode: %s", c.Mode)
	}
```